dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1667,6 +1668,44 @@ public void unstarEntry(@ApiParam(hidden = true) @Parameter(hidden = true, name
         return workflow.getStarredUsers();
     }
 
+    /**
+     *
+     * @param user  The user the workflow belongs to
+     * @param workflowId    The ID of the workflow to change descriptor type
+     * @param descriptorLanguage    The descriptor type to change to
+     * @return  The modified workflow
+     */
+    @POST
+    @Path("/{workflowId}/descriptorType")
+    @Timed
+    @Operation(operationId = "updateDescriptorType", summary = "Changes the descriptor type of an unpublished, invalid workflow.", description = "Use with caution. This deletes all the workflowVersions, only use if there's nothing worth keeping in the workflow.", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiOperation(value = "hidden", hidden = true)
+    public Workflow updateLanguage(
+            @ApiParam(hidden = true) @Parameter(hidden = true, name = "user") @Auth User user,
+            @ApiParam(value = "Workflow to grab starred users for.", required = true) @PathParam("workflowId") Long workflowId,
+            @ApiParam(value = "Descriptor type to update to", required = true) @QueryParam("descriptorType") DescriptorLanguage descriptorLanguage) {
+        Workflow workflow = workflowDAO.findById(workflowId);
+        checkEntry(workflow);
+        if (workflow.getIsPublished()) {
```

Comment:
Agreed that there should be more disallowed conditions. 
Like if you have any versions with properly indexed descriptors (not .dockstore.yml) maybe

Reply:
below checks each version for validity.